ext/cuda/cuda_iterator.jl

```diff
@@ -0,0 +1,151 @@
+
+# This file contains the cuda implementation for `device_iterator.jl` in the `ferrite-addons` module.
+# this iterator can iterates over all cells in the grid or in a subdomain depends on the subdomain index (sdh_idx = -1 -> all cells, otherwise index of the subdomain)
+# it takes in its constructor the `mem_alloc` which is the memory allocation for the entire group and 
+# from it extracts the `cell_mem` which is the memory allocation for each cell (i.e. memory per cell)
+
+# iterator with no memory allocation (for testing purposes)
+function _cell_iterator(dh::DeviceDofHandlerData,sdh_idx::Integer,n_cells::Integer)
```

Comment:
Maybe we can be a bit more expressive here. E.g.
```suggestion
function build_cell_iterator(dh::DeviceDofHandlerData,sdh_idx::Integer,n_cells::Integer)
```